src/main/webapp/pagination.js

```diff
@@ -9,9 +9,152 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.*/
 
-// change content of page 
+
+let fileLength = Number.MAX_VALUE;
+
+// the function contains algo for fetching page 
+// and updating current page's content
 async function changePage(page) {
-    // TODO :- the function contains algo for fetching data 
-    // and updating current page's content
-    // WILL BE ADDED ALONG WITH PAGINATION
+    // change current page (page user is viewing at present) to the 
+    // value provided while calling the function 
+    updateCurrentPage(page);
+    const logs = document.getElementById(LOGS).getAttribute("aria-selected");
+    const searchString = document.getElementById(SEARCH_BAR).value;
+    const fileType = logs == "true" ? LOGS : ERRORS ;
+    // get the page no to be fetched next 
+    const fetchPage = getPageToBeFetched();
+    let fileName = document.getElementById(FILE_NAME).value;
+    // remove blank spaces from fileName
+    fileName =  fileName.trim();
+
+    // if we are on page 1 we need to reset the value of last page 
+    // else we already have requested page in window which we can simply
+    // display 
+    if(page == 1){
```

Comment:
if (page == 1) {
  // Update the window etc
}
// display at the end

Reply:
we need to put display in else  because if page ==1 we don't know if its page 1 of same file which user was viewing previously or its new file . if its new file we don't have a window for it we first need to fetch records and create window and then display but if its not 1st page we know we have window maintained for that file and at that time we can display first and can fetch next page async later